tech-specs/src/explore/design/dkg.md

```diff
@@ -1 +1,154 @@
+posted for a DKG instance includes the session id \\( \tau \\).
+
+This means that the storage of a node must be adapted. Each instance of a
+DKG protocol from Ferveo is a state machine that keeps all necessary data
+from the protocol instance. We thus need to store the following data
+ - A list of sesssion ids from the last completed DKG instance up until the
+   latest started (we may prune ids for failed instances however) 
+ - The DKG state machine associated to each of these session ids.
+
+##  The DKG state machine
+
+We first describe each new message type needed for the protocol.
+- `StartDKG`: A message from the current block proposer to start the protocol
+- `AnnounceStake`: Validators participating in this instance of the DKG
+  broadcast their stake
+- `PartitionShares`: Once the block containing all of the `AnnounceStake`
+  messages are finalized, the set of participants is fixed and the
+  partition of the weight shares is added to the block.
+- `Deal`: This message contains the PVSS transcript from a validator
+  acting as a dealer
+- `Aggregate`: Combine the PVSS instances into a single one, thereby
+  producing the final key.
+
+In Ferveo, the DKG protocol is modelled as a state machine. This allows a
+validator to initialize an instance and drive it forward by reacting to
+each of the above message in the appropriate way. All necessary state
+is persisted by this state machine. We describe the  appropriate action for
+each of the above transactions.
+
+ - `StartDKG`: Initialize a fresh instace of the DKG protocol from Ferveo.
+   Then send out an `AnnounceStake` transaction.
+ - `AnnounceStake`: These simply need to be verified. The current block
+   proposer adds a `PartitionShares` message to the block after the block starting
+   the DKG instance. It is based off all the `AnnounceStake` transactions
+   which will also be included in the next block (for verification).
+ - `PartitionShares`: Validators verify the partitions and weights and
+   should extract their weight shares and store them in the DKG state machine.
+   Then they should send out `Deal` transactions.
+ - `Deal`: These need to be verified. The current block proposer checks if 
+   enough PVSS instances have been reached to meet the security threshold.
+   If so, it adds a `Aggregate` transaction to the block. Otherwise, the
+   next block proposer will take on the job. 
+ - `Aggregate`: This simply needs to be verified and the resulting key added
+   to the DKG instance. 
+
+Most of the above transactions occur as a reaction to other messages
+and/or data in the proposed block header. The three exceptions are the 
+`StartDKG`, `PartitionShares`, and `Aggregate` transactions. These must be 
+made by the current block proposer if their current DKG state machine is in the 
+appropriate state and/or the appropriate conditions are met. Thus block 
+proposers must check the state machines of their active DKG instances.
```

Comment:
So in the simplified model I think it looks like:

if 2/3 weight of Deal transactions has been posted, AND I am the next block producer, then I must include in the next block an Aggregate message (aggregating all previous Deal transactions) OR include a Complaint message (if one or more Deal transactions was invalid)